cocotb/ipython_support.py

```diff
@@ -0,0 +1,89 @@
+# Copyright cocotb contributors
+# Licensed under the Revised BSD License, see LICENSE for details.
+# SPDX-License-Identifier: BSD-3-Clause
+import IPython
+from IPython.terminal.ipapp import load_default_config
+from IPython.terminal.prompts import Prompts, Token
+
+import cocotb
+
+
+class SimTimePrompt(Prompts):
+    """ custom prompt that shows the sim time after a trigger fires """
+    _show_time = 1
+
+    def in_prompt_tokens(self, cli=None):
+        tokens = super().in_prompt_tokens()
+        if self._show_time == self.shell.execution_count:
+            tokens = [
+                (Token.Comment, "sim time: {}".format(cocotb.utils.get_sim_time())),
+                (Token.Text, "\n"),
+            ] + tokens
+        return tokens
+
+
+def _runner(shell, x):
+    """ Handler for async functions """
+    ret = cocotb.scheduler.queue_function(x)
+    shell.prompts._show_time = shell.execution_count
+    return ret
+
+
+async def embed(user_ns: dict = {}):
+    """
+    Start an ipython shell in the current coroutine.
+
+    Unlike using :func:`IPython.embed` directly, the :keyword:`await` keyword
+    can be used directly from the shell to wait for triggers.
+    The :keyword:`yield` keyword from the legacy :ref:`yield-syntax` is not supported.
+
+    This coroutine will complete only when the user exits the interactive session.
+
+    Args:
+        user_ns:
+            The variables to have made available in the shell.
+            Passing ``locals()`` is often a good idea.
+            ``cocotb`` will automatically be included.
+
+    Notes:
```

Comment:
Use a Sphinx ``.. note::``?

Reply:
I think this is part of the `sphinx.napoleon` syntax, but I suspect we don't use it elsewhere. Happy to go either way.